project/src/main/webapp/gmail.js

```diff
@@ -0,0 +1,365 @@
+            });
+            messageRequest.execute(response => {
+                //Convert from base64 encoding to text.
+                const decodedEmail = atob(response.raw.replace(/-/g, '+').replace(/_/g, '/'));
+                const emailBodyValue = this.getEmailBody(decodedEmail);
+                const emailDateValue = this.getEmailDate(decodedEmail);
+                const messageID = this.getMessageID(decodedEmail);
+                if (this.isActualEmail(emailBodyValue, emailDateValue)) {
+                    this.emailObjects[messageID] = { emailDate: emailDateValue, emailBody: emailBodyValue };
+                }
+                this.getEmailObjects();
+                resolve();
+            });
+        });
+        return getMessagePromise;
+    }
+
+    //Retrieve messages using hardcoded queries and the signed-in email.
+    listMessages() {
+        this.emailObjects = {};
+        const listMessagePromise = new Promise((resolve) => {
+            const promiseArray = [];
+            var getPageOfMessages = (request, result) => {
+                request.execute(resp => {
+                    result = result.concat(resp.messages);
+                    var nextPageToken = resp.nextPageToken;
+                    if (nextPageToken) {
+                        request = gapi.client.gmail.users.messages.list({
+                            'userId': 'me',
+                            'pageToken': nextPageToken,
+                            'q': this.question
+                        });
+                        getPageOfMessages(request, result);
+                    } else {
+                        for (var i = 0; i < result.length; i++) {
+                            if (result[i] === undefined) {
+                                console.log("There are no messages from the query");
+                            }
+                            else {
+                                console.log(result.length);
+                                console.log(result);
+                                console.log(result[i].id);
+                                // if (result[i].id === undefined) {
+                                //   this.getMessage(result[i].id);  
+                                // }
+                                promiseArray.push(this.getMessage(result[i].id));
+                            }
+                        }
+                        Promise.all(promiseArray).then(() => resolve()
+                        ); //Promise.all and then this.resolve
+                    }
+                });
+            };
+            var initialRequest = gapi.client.gmail.users.messages.list({
+                'userId': 'me',
+                'q': this.question
+            });
+            getPageOfMessages(initialRequest, []);
+        });
+        console.log("The promise ended");
+        return listMessagePromise;
+    }
+
+    get question() {
+        return this.question;
+    }
+    get totalObjects() {
```

Comment:
This is unused, remove it

Reply:
I think that Brandon's class uses these getters to get the question and the emailObject.

Reply:
Removed these getters!